components/event-publisher-proxy/testing/utils.go

```diff
@@ -163,11 +164,33 @@ func WaitForHandlerToStart(t *testing.T, healthEndpoint string) {
 
 // GeneratePortOrDie generates a random 5 digit port or fail
 func GeneratePortOrDie() int {
-	port, err := generatePort()
-	if err != nil {
-		log.Fatalf("Failed to generate port with error: %v", err)
+	tick := time.NewTicker(time.Second / 2)
+	defer tick.Stop()
+
+	timeout := time.NewTimer(time.Minute)
+	defer timeout.Stop()
+
+	for {
+		select {
+		case <-tick.C:
+			{
+				port, err := generatePort()
```

Comment:
Would it make sense to integrate this retry-timeout-mimik directly into `generatePort() (int, error)`? So all callers would benefit of it.

Reply:
If we integrated the retry loop into the `generatePort`, we'll lose retrying if the port is in use.